File: DSA/saumya/Day11_Q2.py
class Priority_Queue:
    def __init__(self):
        self.queue = []

    def insert(self, item):
        self.queue.append(item)
        self.queue.sort()

    def extract_min(self):
        if len(self.queue) == 0:
            return None
        return self.queue.pop(0)

priority_queue = Priority_Queue()
priority_queue.insert(4)
priority_queue.insert(2)
print(priority_queue.extract_min())  # Output: 2
print(priority_queue.extract_min())  # Output: 4

File: DSA/Yash_Gupta/Day_11_Q2.java
import java.util.*;

class PriorityQueue {
    ArrayList<Integer> heap;
    public PriorityQueue() {
        heap = new ArrayList<Integer>();
    }
    public void insert(int val) {
        heap.add(val);
    }
    public int extractMin() {
        if (heap.isEmpty())
            return -1;
        int min = Integer.MAX_VALUE;
        for (int i: heap) {
            if (i < min)
                min = i;
        }
        heap.remove((Object) min);
        return min;
    }
}

public class Day_11_Q2 {
    public static void main(String[] args) {
        PriorityQueue queue = new PriorityQueue();
        queue.insert(2);
        queue.insert(4);
        System.out.println(queue.extractMin());
        System.out.println(queue.extractMin());
    }
}